<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PasswordGeneratorWidget</class>
 <widget class="QWidget" name="PasswordGeneratorWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>775</width>
    <height>512</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Generate Password</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QGridLayout" name="passwordFieldLayout">
     <item row="0" column="2">
      <widget class="QPushButton" name="buttonGenerate">
       <property name="focusPolicy">
        <enum>Qt::FocusPolicy::TabFocus</enum>
       </property>
       <property name="accessibleDescription">
        <string>Regenerate password</string>
       </property>
       <property name="shortcut">
        <string notr="true">Ctrl+R</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QProgressBar" name="entropyProgressBar">
       <property name="minimumSize">
        <size>
         <width>50</width>
         <height>5</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>5</height>
        </size>
       </property>
       <property name="styleSheet">
        <string notr="true">QProgressBar {
	border: none;
	height: 2px;
    font-size: 1px;
	background-color: transparent;
    padding: 0 1px;
}
QProgressBar::chunk {
	background-color: #c0392b;
	border-radius: 2px;
}</string>
       </property>
       <property name="maximum">
        <number>200</number>
       </property>
       <property name="value">
        <number>100</number>
       </property>
       <property name="textVisible">
        <bool>false</bool>
       </property>
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="invertedAppearance">
        <bool>false</bool>
       </property>
       <property name="textDirection">
        <enum>QProgressBar::Direction::TopToBottom</enum>
       </property>
       <property name="format">
        <string>%p%</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLineEdit" name="editNewPassword"/>
     </item>
     <item row="0" column="3">
      <widget class="QPushButton" name="buttonCopy">
       <property name="focusPolicy">
        <enum>Qt::FocusPolicy::TabFocus</enum>
       </property>
       <property name="toolTip">
        <string>Copy password</string>
       </property>
       <property name="accessibleDescription">
        <string>Copy password</string>
       </property>
       <property name="shortcut">
        <string notr="true">Ctrl+C</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <layout class="QHBoxLayout" name="passwordStrengthTextLayout">
       <item>
        <widget class="QLabel" name="strengthLabel">
         <property name="minimumSize">
          <size>
           <width>70</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>30</height>
          </size>
         </property>
         <property name="text">
          <string comment="Password strength">strength</string>
         </property>
         <property name="textFormat">
          <enum>Qt::TextFormat::PlainText</enum>
         </property>
         <property name="alignment">
          <set>Qt::AlignmentFlag::AlignLeading|Qt::AlignmentFlag::AlignLeft|Qt::AlignmentFlag::AlignTop</set>
         </property>
         <property name="margin">
          <number>3</number>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="passwordLengthLabel">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>70</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>30</height>
          </size>
         </property>
         <property name="text">
          <string>passwordLength</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTop|Qt::AlignmentFlag::AlignTrailing</set>
         </property>
         <property name="margin">
          <number>3</number>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_6">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="entropyLabel">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>70</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>entropy</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTop|Qt::AlignmentFlag::AlignTrailing</set>
         </property>
         <property name="margin">
          <number>3</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="0" column="1">
      <widget class="QPushButton" name="togglePasswordButton">
       <property name="focusPolicy">
        <enum>Qt::FocusPolicy::TabFocus</enum>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="passwordLengthSliderLayout">
     <property name="spacing">
      <number>15</number>
     </property>
     <property name="sizeConstraint">
      <enum>QLayout::SizeConstraint::SetMinimumSize</enum>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QLabel" name="labelLength">
       <property name="text">
        <string>&amp;Length:</string>
       </property>
       <property name="buddy">
        <cstring>spinBoxLength</cstring>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="sliderLength">
       <property name="accessibleName">
        <string>Password length</string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>128</number>
       </property>
       <property name="sliderPosition">
        <number>20</number>
       </property>
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::TickPosition::TicksBelow</enum>
       </property>
       <property name="tickInterval">
        <number>8</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="spinBoxLength">
       <property name="accessibleName">
        <string>Password length</string>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>999</number>
       </property>
       <property name="value">
        <number>20</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="buttonAdvancedMode">
       <property name="focusPolicy">
        <enum>Qt::FocusPolicy::TabFocus</enum>
       </property>
       <property name="toolTip">
        <string>Switch to advanced mode</string>
       </property>
       <property name="text">
        <string>Advanced</string>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">optionButtons</string>
       </attribute>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="minimumSize">
      <size>
       <width>580</width>
       <height>0</height>
      </size>
     </property>
     <property name="title">
      <string>Character Types</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="sizeConstraint">
       <enum>QLayout::SizeConstraint::SetMinimumSize</enum>
      </property>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <layout class="QGridLayout" name="characterButtons">
          <property name="bottomMargin">
           <number>6</number>
          </property>
          <item row="0" column="3">
           <widget class="QPushButton" name="checkBoxSpecialChars">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="focusPolicy">
             <enum>Qt::FocusPolicy::TabFocus</enum>
            </property>
            <property name="toolTip">
             <string>Special characters</string>
            </property>
            <property name="accessibleName">
             <string>Special characters</string>
            </property>
            <property name="text">
             <string notr="true">/ * + &amp;&amp; …</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">optionButtons</string>
            </attribute>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="checkBoxQuotes">
            <property name="focusPolicy">
             <enum>Qt::FocusPolicy::TabFocus</enum>
            </property>
            <property name="toolTip">
             <string>Quotes</string>
            </property>
            <property name="accessibleName">
             <string>Quotes</string>
            </property>
            <property name="text">
             <string notr="true">&quot; '</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">optionButtons</string>
            </attribute>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="checkBoxPunctuation">
            <property name="focusPolicy">
             <enum>Qt::FocusPolicy::TabFocus</enum>
            </property>
            <property name="toolTip">
             <string>Punctuation</string>
            </property>
            <property name="accessibleName">
             <string>Punctuation</string>
            </property>
            <property name="text">
             <string notr="true">. , : ;</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">optionButtons</string>
            </attribute>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QPushButton" name="checkBoxDashes">
            <property name="focusPolicy">
             <enum>Qt::FocusPolicy::TabFocus</enum>
            </property>
            <property name="toolTip">
             <string>Dashes and Slashes</string>
            </property>
            <property name="accessibleName">
             <string>Dashes and Slashes</string>
            </property>
            <property name="text">
             <string notr="true">\ / | _ -</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">optionButtons</string>
            </attribute>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QPushButton" name="checkBoxUpper">
            <property name="focusPolicy">
             <enum>Qt::FocusPolicy::TabFocus</enum>
            </property>
            <property name="toolTip">
             <string>Upper-case letters</string>
            </property>
            <property name="accessibleName">
             <string>Upper-case letters</string>
            </property>
            <property name="text">
             <string notr="true">A-Z</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">optionButtons</string>
            </attribute>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QPushButton" name="checkBoxNumbers">
            <property name="focusPolicy">
             <enum>Qt::FocusPolicy::TabFocus</enum>
            </property>
            <property name="toolTip">
             <string>Numbers</string>
            </property>
            <property name="accessibleName">
             <string>Numbers</string>
            </property>
            <property name="text">
             <string notr="true">0-9</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">optionButtons</string>
            </attribute>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QPushButton" name="checkBoxLower">
            <property name="focusPolicy">
             <enum>Qt::FocusPolicy::TabFocus</enum>
            </property>
            <property name="toolTip">
             <string>Lower-case letters</string>
            </property>
            <property name="accessibleName">
             <string>Lower-case letters</string>
            </property>
            <property name="text">
             <string notr="true">a-z</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">optionButtons</string>
            </attribute>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QPushButton" name="checkBoxMath">
            <property name="focusPolicy">
             <enum>Qt::FocusPolicy::TabFocus</enum>
            </property>
            <property name="toolTip">
             <string>Math Symbols</string>
            </property>
            <property name="accessibleName">
             <string>Math Symbols</string>
            </property>
            <property name="text">
             <string notr="true">&lt; &gt; * + ! ? =</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">optionButtons</string>
            </attribute>
           </widget>
          </item>
          <item row="0" column="4">
           <widget class="QPushButton" name="checkBoxExtASCII">
            <property name="focusPolicy">
             <enum>Qt::FocusPolicy::TabFocus</enum>
            </property>
            <property name="toolTip">
             <string>Extended ASCII</string>
            </property>
            <property name="accessibleName">
             <string>Extended ASCII</string>
            </property>
            <property name="text">
             <string>Extended ASCII</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">optionButtons</string>
            </attribute>
           </widget>
          </item>
          <item row="1" column="4">
           <widget class="QPushButton" name="checkBoxBraces">
            <property name="focusPolicy">
             <enum>Qt::FocusPolicy::TabFocus</enum>
            </property>
            <property name="toolTip">
             <string>Braces</string>
            </property>
            <property name="accessibleName">
             <string>Braces</string>
            </property>
            <property name="text">
             <string notr="true">{ [ ( ) ] }</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">optionButtons</string>
            </attribute>
           </widget>
          </item>
          <item row="0" column="5">
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QWidget" name="advancedContainer" native="true">
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <layout class="QGridLayout" name="gridLayout">
             <property name="bottomMargin">
              <number>6</number>
             </property>
             <item row="1" column="0">
              <widget class="QLabel" name="labelExcludedChars">
               <property name="text">
                <string>Do not include:</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLineEdit" name="editAdditionalChars">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>375</width>
                 <height>0</height>
                </size>
               </property>
               <property name="toolTip">
                <string>Additional characters to use for the generated password</string>
               </property>
               <property name="accessibleName">
                <string>Additional characters</string>
               </property>
               <property name="clearButtonEnabled">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item row="1" column="2">
              <widget class="QPushButton" name="buttonAddHex">
               <property name="focusPolicy">
                <enum>Qt::FocusPolicy::TabFocus</enum>
               </property>
               <property name="toolTip">
                <string>Add non-hex letters to &quot;do not include&quot; list</string>
               </property>
               <property name="accessibleName">
                <string>Hex Passwords</string>
               </property>
               <property name="text">
                <string>Hex</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLineEdit" name="editExcludedChars">
               <property name="sizePolicy">
                <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>375</width>
                 <height>0</height>
                </size>
               </property>
               <property name="toolTip">
                <string>Character set to exclude from generated password</string>
               </property>
               <property name="accessibleName">
                <string>Excluded characters</string>
               </property>
               <property name="clearButtonEnabled">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="label">
               <property name="text">
                <string>Also choose from:</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QCheckBox" name="checkBoxExcludeAlike">
             <property name="toolTip">
              <string>Excluded characters: &quot;0&quot;, &quot;1&quot;, &quot;l&quot;, &quot;I&quot;, &quot;O&quot;, &quot;|&quot;, &quot;﹒&quot;, &quot;B&quot;, &quot;8&quot;, &quot;G&quot;, &quot;6&quot;</string>
             </property>
             <property name="accessibleDescription">
              <string>Excluded characters: &quot;0&quot;, &quot;1&quot;, &quot;l&quot;, &quot;I&quot;, &quot;O&quot;, &quot;|&quot;, &quot;﹒&quot;, &quot;B&quot;, &quot;8&quot;, &quot;G&quot;, &quot;6&quot;</string>
             </property>
             <property name="text">
              <string>Exclude look-alike characters</string>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">optionButtons</string>
             </attribute>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="checkBoxEnsureEvery">
             <property name="text">
              <string>Pick characters from every group</string>
             </property>
             <attribute name="buttonGroup">
              <string notr="true">optionButtons</string>
             </attribute>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <item>
      <spacer name="horizontalSpacer_5">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="buttonApply">
       <property name="text">
        <string>Apply</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="buttonClose">
       <property name="text">
        <string>Close</string>
       </property>
       <property name="shortcut">
        <string>Esc</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>buttonGenerate</tabstop>
  <tabstop>buttonCopy</tabstop>
  <tabstop>sliderLength</tabstop>
  <tabstop>spinBoxLength</tabstop>
  <tabstop>buttonAdvancedMode</tabstop>
  <tabstop>buttonClose</tabstop>
 </tabstops>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="optionButtons">
   <property name="exclusive">
    <bool>false</bool>
   </property>
  </buttongroup>
 </buttongroups>
</ui>
